src/xunit.performance.api/Profilers/Etw/PerformanceMonitorCounter.cs

```diff
@@ -0,0 +1,106 @@
+        /// <param name="name">Performance monitor counter name.</param>
+        /// <param name="unit">Performance monitor counter unit.</param>
+        /// <param name="id">Performance monitor counter Id.</param>
+        public PerformanceMonitorCounter(string displayName, string name, string unit, int id)
+        {
+            DisplayName = displayName;
+            Name = name;
+            Unit = unit;
+            Id = id;
+        }
+
+        /// <summary>
+        /// Performance monitor counter Id.
+        /// </summary>
+        public int Id { get; }
+
+        /// <summary>
+        /// Performance monitor counter friendly name.
+        /// </summary>
+        public string DisplayName { get; }
+
+        /// <summary>
+        /// Performance monitor counter name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Performance monitor counter unit.
+        /// </summary>
+        public string Unit { get; }
+
+        /// <summary>
+        /// Serves as the default hash function.
+        /// </summary>
+        /// <returns>A hash code for the current object.</returns>
+        public override int GetHashCode()
+        {
+            return Id ^ DisplayName.GetHashCode() ^ Name.GetHashCode() ^ Unit.GetHashCode();
+        }
+
+        /// <summary>
+        /// Determines whether the specified object is equals to this object.
+        /// </summary>
+        /// <param name="obj">The object to compare with this object.</param>
+        /// <returns>True if the specified object is equal to the current object; otherwise, false.</returns>
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+                return false;
+
+            var pmc = obj as PerformanceMonitorCounter;
+            if (pmc == null)
+                return false;
+
+            return Id == pmc.Id
+                && DisplayName == pmc.DisplayName
+                && Name == pmc.Name
+                && Unit == pmc.Unit;
+        }
+
+        /// <summary>
+        /// Determines whether two specified PerformanceMonitorCounter objects have the same value.
+        /// </summary>
+        /// <param name="lhs">The first PerformanceMonitorCounter to compare, or null.</param>
+        /// <param name="rhs">The second PerformanceMonitorCounter to compare, or null.</param>
+        /// <returns>True if its two operands refer to the same object or if the values of its operands are equal; otherwise, false.</returns>
+        public static bool operator ==(PerformanceMonitorCounter lhs, PerformanceMonitorCounter rhs)
+        {
+            if (ReferenceEquals(lhs, rhs))
+                return true;
+
+            if ((object)lhs == null || (object)rhs == null)
+                return false;
+
+            return lhs.Id == rhs.Id
+                && lhs.DisplayName == rhs.DisplayName
+                && lhs.Name == rhs.Name
```

Comment:
Same question here.

Reply:
Same as above. I'll clean it up.

---
In reply to: [146052047](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r146052047) [](ancestors = 146052047)